src/node.cc

```diff
@@ -543,7 +543,7 @@ void RegisterSignalHandler(int signal,
   struct sigaction sa;
   memset(&sa, 0, sizeof(sa));
   sa.sa_sigaction = handler;
-  sa.sa_flags = reset_handler ? SA_RESETHAND : 0;
+  sa.sa_flags = reset_handler ? SA_RESETHAND : SA_SIGINFO;
```

Comment:
`SA_SIGINFO` should always be set, regardless of `reset_handler`, right?

Reply:
Thanks. Updated and added an assertion to the existing `register-signal-handler` addon test to check that the handler function is called with three arguments.